plugins/versioning/src/main/kotlin/versioning/PreviousDocumentationCopier.kt

```diff
@@ -0,0 +1,56 @@
+package versioning
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.jetbrains.dokka.base.renderers.PostAction
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.plugability.plugin
+import org.jetbrains.dokka.plugability.query
+import org.jetbrains.dokka.plugability.querySingle
+import org.jetbrains.dokka.templates.TemplateProcessingStrategy
+import org.jetbrains.dokka.templates.TemplatingPlugin
+import org.jetbrains.dokka.versioning.VersioningConfiguration
+import org.jetbrains.dokka.versioning.VersioningPlugin
+import java.io.File
+
+class DefaultPreviousDocumentationCopier(val context: DokkaContext) : PostAction {
+    private val versioningHandler by lazy { context.plugin<VersioningPlugin>().querySingle { versioningHandler } }
+    private val processingStrategies: List<TemplateProcessingStrategy> =
+        context.plugin<TemplatingPlugin>().query { templateProcessingStrategy }
+
+    override fun invoke() {
+        versioningHandler.previousVersions.onEach { (_, dirs) -> copyVersion(dirs.src, dirs.dst) }
```

Comment:
I use the version source to resolve a link in `VersionsNavigationCreator `.  I check is there a needed file in the corresponding  version source folder to avoid generating the invalid links (In the navigator such version will be strikethrough). At the copying (also creating navigator) moment the destination folder can't be used since not all version folder are copied.